typescript-generator-core/src/main/java/cz/habarta/typescript/generator/emitter/TsEnumModel.java

```diff
@@ -12,7 +12,7 @@
 
     private final EnumKind kind;
     private final List<EnumMemberModel> members;
-    private boolean isConstEnum = true;
+    private boolean isNonConstEnum = false;
```

Comment:
could you please mark this field `final` and make this class immutable again?

Reply:
About the final thing, it would break backwards-compatibility, since I cannot use getters/setters any more, but I'll need to pass the field to the constructor.

Reply:
I think we can have two constructors: one for backward compatibility and one new with all the fields.

And instead of setter we can have `withIsNonConstEnum(boolean)` or just `asNonConst()` that would return new instance.